prog.go

```diff
@@ -511,6 +503,52 @@ func newProgramWithOptions(spec *ProgramSpec, opts ProgramOptions) (*Program, er
 	return nil, internal.ErrorWithLog("load program", err, logBuf)
 }
 
+func retryLogAttrs(attr *sys.ProgLoadAttr, startSize uint32, err error) bool {
+	// ENOSPC means the log was enabled on the previous iteration, so we only
+	// need to grow the buffer.
+	if errors.Is(err, unix.ENOSPC) {
+		if attr.LogTrueSize != 0 {
+			// Kernel supports LogTrueSize and previous iteration undershot the buffer
+			// size. Try again with the given true size.
+			attr.LogSize = attr.LogTrueSize
+			return true
+		}
+
+		// ENOSPC means we've loaded the program before and the log buffer was too
+		// small.
+		if attr.LogSize == maxVerifierLogSize {
```

Comment:
I'm not sure you will ever reach this condition, because I don't think the kernel will return `ENOSPC` when using the max verifier log size. That said, it doesn't hurt to be defensive here.

Reply:
Good point, let's not assume ENOSPC will be returned. If the max log size is hit, we should never retry.